File: 965-unique-email-addresses/unique-email-addresses.java
class Solution {

    public String getSanitizedEmail(String email){
        String sanitizedEmail = "";
        for(int i = 0 ; i < email.length() ; i++){
            char c = email.charAt(i);
            if(c == '+' || c == '@')break;
            if(c == '.')continue;
            sanitizedEmail+=c;
        }
        int indexOfAt = email.indexOf('@');
        sanitizedEmail = sanitizedEmail + email.substring(indexOfAt);
        return sanitizedEmail;
    }

    public int numUniqueEmails(String[] emails) {
        Set<String> s = new HashSet<>();
        for(int i = 0 ; i < emails.length ; i++){
           s.add(getSanitizedEmail(emails[i]));
        }
        return s.size();
    }
}

File: 965-unique-email-addresses/unique-email-addresses.cpp
class Solution {
public:
    string getSanitizedEmail(string email){
       string sanitizedEmail = "";
       for(int i = 0 ; i < email.size() ; i++){
            if(email[i] == '@' || email[i] == '+')break;
            if(email[i] == '.')continue;
            sanitizedEmail+=email[i];
       }
       string domainName = email.substr(email.find("@"));
       sanitizedEmail = sanitizedEmail + "@" + domainName;
       return  sanitizedEmail;
    }
    int numUniqueEmails(vector<string>& emails) {
        unordered_set<string> s;
        for(int i = 0 ; i < emails.size() ; i++){
            s.insert(getSanitizedEmail(emails[i]));
        }
        return s.size();
    }
};